Statistics epic 3of3

Depechist opened the pull request:
- Полностью готовый Эпик Статистики

Depechist committed:
UI + Network

- Сверстан экран коллекции NFT
- Добавлен слой для обновления UI данными из сети
- Добавлена навигация между экранами

Depechist committed:
Fix

- Добавлен алерт на экране Пользователя в случае попытки открыть пустую коллекцию NFT
- Добавлена локализация тайтла на экране Коллекция NFT

Nikitaaeee commented on FakeNFT/Scenes/Common/Views/CollectionCell/CollectionCell.swift:
```diff
@@ -0,0 +1,156 @@
+        likeButton.snp.makeConstraints { make in
+            make.top.equalToSuperview()
+            make.right.equalToSuperview()
+            make.width.equalTo(40)
+            make.height.equalTo(likeButton.snp.width)
```
Этот код можно немного сократить, например, написав в одну скоку:
`make.width.height.equalTo(40)`
SnapKit позволяет объединять аргументы через точку для того, чтобы затем задать им одно значение констрейнта.

Также можно написать так:
`make.size.equalTo(40)`
Это более краткая версия прошлого варианта, так как **size** устанавливает _ширину_ и _высоту_.

Nikitaaeee commented on FakeNFT/Scenes/Common/Views/CollectionCell/CollectionCell.swift:
```diff
@@ -0,0 +1,156 @@
+//  Created by Artem Adiev on 15.11.2023.
+//
+
+import UIKit
+
+
+final class CollectionCell: UICollectionViewCell, ReuseIdentifying {
```
Можно убрать одну лишнюю пустую строку после импорта :)

Nikitaaeee commented on FakeNFT/Scenes/Statistics/NFTCollection/CollectionViewModel.swift:
```diff
@@ -0,0 +1,64 @@
+            }
+        }
+        // Когда все запросы завершены, скрываем индикатор загрузки
+        fetchGroup.notify(queue: .main) {
+            self.hideLoading?()
+        }
```
🔥🔥🔥

Depechist committed:
Sorting in UserDefaults

- Теперь метод сортировки таблицы сохраняется в UserDefaults. По умолчанию - по рейтингу. Далее - по выбору пользователя.

Nikitaaeee commented on FakeNFT/Scenes/Statistics/Rating/RatingViewModel.swift:
```diff
@@ -7,6 +7,10 @@
 
 import Foundation
 
+struct UserDefaultsKeys {
+    static let sortingOrder = "sortingOrder"
+}
+
 final class RatingViewModel {
     // MARK: - Properties
```
Хотя использование **UserDefaults** напрямую во вью модели работает, это может затруднить тестирование и поддержку кода. 
Лучше создать отдельный _сервис_ или _менеджер_ для управления хранилищем **UserDefaults**. 
Это улучшит модульность и тестируемость кода :)
